flask/app/models.py

```diff
@@ -450,17 +464,29 @@ class Variant(db.Model):
     polyphen_score: int = db.Column(db.Float, nullable=True)
     cadd_score: int = db.Column(db.Float, nullable=True)
     gnomad_af: int = db.Column(db.Float, nullable=True)
+    __table_args__ = (db.UniqueConstraint("variant_id", "analysis_id"),)
 
 
-class AnalyzedVariant(db.Model):
+@dataclass
+class Genotype(db.Model):
     variant_id: int = db.Column(
         db.Integer, db.ForeignKey("variant.variant_id"), primary_key=True
     )
     analysis_id: int = db.Column(
-        db.Integer, db.ForeignKey("analysis.analysis_id"), primary_key=True
+        db.Integer,
+        db.ForeignKey("analysis.analysis_id"),
+        primary_key=True,
     )
-    analyses = db.relationship("Analysis", backref="variant")
-    variant = db.relationship("Variant", backref="analysis")
+    dataset_id: int = db.Column(
+        db.Integer,
+        db.ForeignKey("dataset.dataset_id"),
+        primary_key=True,
+    )
+
+    variants = db.relationship("Variant", backref="genotype")
+    analyses = db.relationship("Analysis", backref="genotype")
+    datasets = db.relationship("Dataset", backref="genotype")
```

Comment:
Examining the schematic, these should all be singular. I don't know if the backref really makes sense as they will be aggregations, e.g.:

the Variant backref would have the zygosities for all datasets with this variant in this analysis
the Analysis backref would have _all_ the zygosities computed for all variants found for all datasets in this analysis
the Dataset backref would have all zygosities across all variants found in all analyses including this variant

They might be useful, since they won't be eager loaded anyway, for performing specific queries across these relationships though.